controllers/replication/secret_controller.go

```diff
@@ -474,16 +479,17 @@ func calculateTargetLabels(originalLabels map[string]string, target api.Replicat
 	return out
 }
 
-func wouldBeInfinite(origin corev1.Secret, rsec api.ReplicatedSecret, target api.ReplicationTarget) bool {
-	computedLabels := calculateTargetLabels(origin.Labels, target)
-	for k, v := range rsec.Spec.Selector.MatchLabels {
-		if computedLabels[k] != v {
-			return false
-		}
+func wouldBeInfinite(origin corev1.Secret, rsec api.ReplicatedSecret, target api.ReplicationTarget) (bool, error) {
+	computedLabels := labels.Set(calculateTargetLabels(origin.Labels, target))
+	selector, err := metav1.LabelSelectorAsSelector(rsec.Spec.Selector)
+	if err != nil {
+		return true, err
```

Comment:
[nitpick] We already have `selector` from line 174 in the main method, we can pass it as an argument to `wouldBeInfinite` instead of rebuilding it here (and then remove the `rsec` argument).